src/main/java/net/dv8tion/jda/api/entities/ThreadChannel.java

```diff
@@ -71,119 +136,319 @@ default GuildMessageChannel getParentMessageChannel()
         throw new UnsupportedOperationException("Parent of this thread is not a MessageChannel. Parent is type: " + getParentChannel().getType().getId());
     }
 
+    /**
+     * Gets the self member, as a member of this thread.
+     *
+     * <br>If the current account is not a member of this thread, this will return null.
+     *
+     * @return The self member of this thread, null if the current account is not a member of this thread.
+     */
     @Nullable
     default ThreadMember getSelfThreadMember()
     {
         return getThreadMember(getJDA().getSelfUser());
     }
 
-    //Only have access to this with GUILD_MEMBERS
+
+    /**
+     * Gets a List of all {@link ThreadMember members} of this thread.
+     * <br>
+     * <br><b>Requirements</b>:
+     * <ul>
+     *     <li>the {@link net.dv8tion.jda.api.requests.GatewayIntent#GUILD_MEMBERS} intent must be enabled.</li>
+     * </ul>
+     *
+     * @return a List of all {@link ThreadMember members} of this thread. This list may be empty, but not null.
+     */
     @Nonnull
     List<ThreadMember> getThreadMembers();
 
+    /**
+     * Gets a {@link ThreadMember} of this thread by their {@link Member}.
+     *
+     * @param member
+     *        The member to get the {@link ThreadMember} for.
+     *
+     * @return The {@link ThreadMember} of this thread for the given member.
+     */
+    //TODO-v5: docs - how much of this relies on the GUILD_MEMBERS intent. Current impl is unfinished and so cannot document.
```

Comment:
This returns null with the same rules as when thread members are present in `getThreadMembers()`.
i.e: Quite a lot.